File: java-samples/Features/attachment-sendfile/contracts/src/main/java/net/corda/samples/sendfile/states/InvoiceState.java
package net.corda.samples.sendfile.states;

import net.corda.core.contracts.BelongsToContract;
import net.corda.core.contracts.ContractState;
import net.corda.core.identity.AbstractParty;
import net.corda.samples.sendfile.contracts.InvoiceContract;
import org.jetbrains.annotations.NotNull;

import java.util.List;

// *********
// * State *
// *********
@BelongsToContract(InvoiceContract.class)
public class InvoiceState implements ContractState {
    private final String invoiceAttachmentID;
    private final List<AbstractParty> participants;

    public InvoiceState(String invoiceAttachmentID, List<AbstractParty> participants) {
        this.invoiceAttachmentID = invoiceAttachmentID;
        this.participants = participants;
    }

    @NotNull
    public String getInvoiceAttachmentID() {
        return invoiceAttachmentID;
    }

    @NotNull
    @Override
    public List<AbstractParty> getParticipants() {
        return participants;
    }
}

File: kotlin-samples/Features/attachment-sendfile/contracts/src/main/kotlin/net/corda/samples/sendfile/states/InvoiceState.kt
package net.corda.samples.sendfile.states

import net.corda.samples.sendfile.contracts.InvoiceContract
import net.corda.core.contracts.BelongsToContract
import net.corda.core.contracts.ContractState
import net.corda.core.identity.AbstractParty

// *********
// * State *
// *********
@BelongsToContract(InvoiceContract::class)
data class InvoiceState(
        val invoiceAttachmentID: String,
        override val participants: List<AbstractParty> = listOf()) : ContractState
